test/Label.test.js

```diff
@@ -418,6 +435,686 @@ describe('Network Label', function() {
+     */
+    var fontOption = (index) => {
+      var nodes = network.body.nodes;
+      return nodes[index].labelModule.fontOptions;
+    };
+
+
+    /**
+     * Helper function for easily accessing bold options in a node
+     */
+    var modBold = (index) => {
+      return fontOption(index).bold;
+    };
+
+
+    var dataNodes = [
+      {id: 0, label: '<b>0</b>'},
+      {id: 1, label: '<b>1</b>'},
+      {id: 2, label: '<b>2</b>', group: 'group1'},
+      {id: 3, label: '<b>3</b>',
+        font: {
+          bold: { color: 'green' },
+        }
+      },
+      {id: 4, label: '<b>4</b>', group: 'group1',
+        font: {
+          bold: { color: 'green' },
+        }
+      },
+    ];
+  
+    // create a network
+    var container = document.getElementById('mynetwork');
+    var data = {
+        nodes: new vis.DataSet(dataNodes),
+        edges: []
+    };
+  
+    var options = {
+      nodes: {
+        font: {
+          multi: true
+        }
+      },
+      groups: {
+        group1: {
+          font: { color: 'red' },
+        },
+        group2: {
+          font: { color: 'white' },
+        },
+      },
+    };
+  
+    var network = new vis.Network(container, data, options);
+
+
+    assert.equal(modBold(0).color, '#343434');  // Default value
+    assert.equal(modBold(1).color, '#343434');  // Default value
+    assert.equal(modBold(2).color, 'red');      // Group value overrides default
+    assert.equal(modBold(3).color, 'green');    // Local value overrides default
+    assert.equal(modBold(4).color, 'green');    // Local value overrides group
+
+
+    //
+    // Change some node values dynamically
+    //
+    data.nodes.update([
+      {id: 1, group: 'group2'},
+      {id: 4, font: { bold: { color: 'orange'}}},
+    ]);
+
+    assert.equal(modBold(0).color, '#343434');  // unchanged
+    assert.equal(modBold(1).color, 'white');    // new group value
+    assert.equal(modBold(3).color, 'green');    // unchanged
+    assert.equal(modBold(4).color, 'orange');   // new local value
+
+    //
+    // Change group options dynamically
+    //
+    network.setOptions({
+      groups: {
+        group1: {
+          font: { color: 'brown' },
+        },
+      },
+    });
+
+
+    assert.equal(modBold(0).color, '#343434');  // unchanged
+    assert.equal(modBold(1).color, 'white');    // Unchanged
+    assert.equal(modBold(2).color, 'brown');    // New group values
+    assert.equal(modBold(3).color, 'green');    // unchanged
+    assert.equal(modBold(4).color, 'orange');   // unchanged
+
+
+    network.setOptions({
+      nodes: {
+        font: {
+          multi: true,
+          bold: {
+            color: 'black'
+          }
+        }
+      },
+    });
+
+    assert.equal(modBold(0).color, 'black');    // nodes default
+    assert.equal(modBold(1).color, 'black');    // more specific bold value overrides group value
+    assert.equal(modBold(2).color, 'black');    // idem
```

Comment:
Idem?

Reply:
Why are these three changed, and the others aren't? I'm missing something here...

Reply:
['Idem'](http://www.dictionary.com/browse/idem) : Same as previous comment in this column.

Reply:
Overview:
- The priority of the options are:  `node, group, default node, global default`.
- `black` is in 'default node'.
- multi-font options, if defined, take precedence. The main font options are fall-backs for the multi-font options.

Specific:
- Node 0 has no 'node' or 'group' options, hence uses 'default node'.
- Nodes 1 has `group2`, which has a default font but no explicit bold font. The bold definition in 'default node' trumps this.
- Node 2 has `group1`, the same logic as with Node 1 applies here.
- Nodes 3 and 4 have bold fonts defined in the node instance. These trump everything.

Welcome to the wonderful world of multi-font options. Precisely this kind of thing is what made fixing the issue so hard. My head still hurts.